### PR TITLE
Convert tests to TypeScript

### PR DESCRIPTION
**Pull request by dstaley**
This PR converts the tests for the existing TypeScript files to TypeScript. Writing the tests in TypeScript gives us the ability to test functionality and the type system. It allows us to ensure that a given change doesn't break compilation on existing TypeScript code.

Moving forward, I'll convert the tests to TypeScript in the same PR as the main file conversion.

### CODE REVIEW

File: src/endpoint.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ const ENDPOINTS = {
     `${ENDPOINTS.bucket(bucket)}/collections` + (coll ? `/${coll}` : ""),
   group: (bucket: string, group?: string) =>
     `${ENDPOINTS.bucket(bucket)}/groups` + (group ? `/${group}` : ""),
-  record: (bucket: string, coll: string, id?: string) =>
+  record: (bucket: string, coll: string, id?: string | number) =>
```
Comment (leplatrem) [inline src/endpoint.ts]: I wouldn't allow number here

File: src/types.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ export interface KintoRequest {
 
 export interface KintoRecord {
   id: string;
-  last_modified: number;
+  last_modified?: number;
```
Comment (leplatrem) [inline src/types.ts]: If the record comes from the server, it will always have this field. Which part of the tests required this?

Comment (dstaley) [reply src/types.ts]: Particularly when updating a record, we know the ID, but we don't pass in a `last_modified` inside the record. I've added a new type, `KintoIdRecord`, which is used when we know a record's ID, but don't want to specify a `last_modified`. `KintoRecord` now extends from `KintoIdRecord`, adding the `last_modified` property.

**Comment by leplatrem** [comment]
Awesome! Thanks!

**Pull request merged**